From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: booky10 <boooky10@gmail.com>
Date: Wed, 12 Jul 2023 19:06:02 +0200
Subject: [PATCH] Add aggressive mob API

Moves the methods from Zombie to Mob, as more Mobs can raise their arm(s)

diff --git a/src/main/java/org/bukkit/entity/Mob.java b/src/main/java/org/bukkit/entity/Mob.java
index 9e289e92ad76cf61d445b77e3765b0e3ea1abcf7..10136b42381a1d6ade68095d5cc6208c53f18df4 100644
--- a/src/main/java/org/bukkit/entity/Mob.java
+++ b/src/main/java/org/bukkit/entity/Mob.java
@@ -187,4 +187,30 @@ public interface Mob extends LivingEntity, Lootable {
      */
     public int getPossibleExperienceReward();
     // Paper end
+
+    // Paper start
+    /**
+     * Some mobs will raise their arm(s) when aggressive:
+     * <ul>
+     *     <li>{@link Drowned}</li>
+     *     <li>{@link Piglin}</li>
+     *     <li>{@link Skeleton}</li>
+     *     <li>{@link Zombie}</li>
+     *     <li>{@link ZombieVillager}</li>
+     *     <li>{@link Illusioner}</li>
+     *     <li>{@link Vindicator}</li>
+     *     <li>{@link Panda}</li>
+     *     <li>{@link Pillager}</li>
+     *     <li>{@link PiglinBrute}</li>
+     * </ul>
+     *
+     * @return wether the mob is aggressive or not
+     */
+    boolean isAggressive();
+
+    /**
+     * @see #isAggressive()
+     */
+    void setAggressive(boolean aggressive);
+    // Paper end
 }
